airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -280,6 +277,149 @@ def test_secret_is_properly_marked(self, connector_spec_dict: dict, detailed_log
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        errors = []
+        for type_path, type in dpath.util.search(specification, "**/type", yielded=True):
+            property_definition = self._get_parent(specification, type_path)
+            if type != "array":
+                # unrelated "items", not an array definition
+                continue
+            items_value = property_definition.get("items", None)
+            if items_value is None:
+                continue
+            elif isinstance(items_value, List):
+                errors.append(f"{type_path} is not just a single item type: {items_value}")
+            elif items_value.get("type") not in ["object", "string"] and "enum" not in items_value:
+                errors.append(f"Items of {type_path} has to be either object or string or define an enum")
+        if len(errors) > 0:
+            pytest.fail("\n".join(errors))
+
+    def test_forbidden_complex_types(self, connector_spec_dict: dict):
+        """
+        not, anyOf, patternProperties, prefixItems, allOf, if, then, else, dependentSchemas and dependentRequired are not allowed
+        """
+        forbidden_keys = [
+            "not",
+            "anyOf",
+            "patternProperties",
+            "prefixItems",
+            "allOf",
+            "if",
+            "then",
+            "else",
+            "dependentSchemas",
+            "dependentRequired",
+        ]
+        specification = connector_spec_dict["connectionSpecification"]
+        found_keys = set()
+        for forbidden_key in forbidden_keys:
+            for path, value in dpath.util.search(specification, f"**/{forbidden_key}", yielded=True):
+                found_keys.add(path)
+
+        for forbidden_key in forbidden_keys:
+            # remove forbidden keys if they are used as properties directly
+            for path, _value in dpath.util.search(specification, f"**/properties/{forbidden_key}", yielded=True):
+                found_keys.remove(path)
+
+        if len(found_keys) > 0:
+            key_list = ", ".join(found_keys)
+            pytest.fail(f"Found the following disallowed JSON schema features: {key_list}")
+
+    def test_date_pattern(self, connector_spec_dict: dict, detailed_logger):
+        """
+        Properties with format date or date-time should always have a pattern defined how the date/date-time should be formatted
+        that corresponds with the format the datepicker component is creating.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        for format_path, format in dpath.util.search(specification, "**/format", yielded=True):
+            if not isinstance(format, str):
+                # format is not a format definition here but a property named format
+                continue
+            if format == "date" or format == "date-time":
```

Comment:
nit: this if statement is superfluous since we compare the format string on line 393 and 397